### PR TITLE
[Bug] fix BERT weight load transpose bug

### LINKED ISSUE #2249: [Bug] Fix transposed BERT pooler weights in TraditionalBertClassifier
### Describe the bug

In `candle-binding`, `TraditionalBertClassifier` loads the BERT pooler weight (`bert.pooler.dense.weight`) transposed, while the classification head in the same constructor loads its weight without transpose. Only one convention can match HuggingFace. The pooler load is the inconsistent one, so the pooled `[CLS]` representation is computed with a transposed weight matrix and the resulting class probabilities / confidence scores are miscalibrated for the traditional BERT sequence-classification path.

In `candle-binding/src/model_architectures/traditional/bert.rs`, `TraditionalBertClassifier::new`:

```rust
// pooler: transposed
let pooler_weight = vb.get(
    (config.hidden_size, config.hidden_size),
    "bert.pooler.dense.weight",
)?;
let pooler_bias = vb.get(config.hidden_size, "bert.pooler.dense.bias")?;
Linear::new(pooler_weight.t()?, Some(pooler_bias));   // .t()

// classifier head a few lines below: NOT transposed
let classifier_weight = vb.get((num_classes, config.hidden_size), "classifier.weight")?;
let classifier_bias = vb.get(num_classes, "classifier.bias")?;
Linear::new(classifier_weight, Some(classifier_bias)); // no .t()
```

HuggingFace stores `nn.Linear` weights as `[out_features, in_features]`, and `candle_nn::Linear::forward(x)` already computes `x @ weight.t()`, so the correct load is as-is (what the classifier head does). The pooler passes `pooler_weight.t()` (`[in, out]`), so forward effectively computes `x @ W` instead of `x @ W^T`. Because the pooler matrix is square it does not raise a shape error; it silently applies the wrong linear map.

### To Reproduce

1. Load a BERT sequence-classification checkpoint through `TraditionalBertClassifier`.
2. Run a forward pass and read the pooled output / logits / softmax confidence.
3. Compare against HuggingFace `transformers` for the same checkpoint and input.
4. The pooled output and confidences diverge because the pooler weight is applied transposed.

### Expected behavior

Pooled output, logits, and confidences match HuggingFace `transformers` for the same checkpoint and input.

### Affected layer

signal

### Additional context

Pre-existing and unrelated to in-flight feature work; noticed while implementing the cross-encoder reranker (PR #2236 / #2247), which reuses the same BERT + pooler + classifier pattern.

Suggested fix: drop the `.t()` so the pooler matches the classifier head and the HuggingFace convention:

```rust
Linear::new(pooler_weight, Some(pooler_bias));
```

and validate by comparing pooled output / logits against HuggingFace `transformers`. Happy to send a small PR with the one-line fix plus a parity check.

### PR DESCRIPTION
**Pull request by Alanxtl**
Closes #2249

## Purpose

Fix incorrect BERT weight orientation in `candle-binding`.

`candle_nn::Linear` expects weights in `[out_features, in_features]` layout and applies the transpose during `forward`. This change removes the extra transpose from:

- `TraditionalBertClassifier` pooler
- `LoRABertClassifier` pooler
- `LoRABertClassifier` task classification head

Added deterministic non-symmetric and non-square matrix tests in the corresponding `*_test.rs` files.

## Test Plan

- `cargo test --manifest-path candle-binding/Cargo.toml --no-default-features --lib test_bert_pooler_uses_huggingface_weight_layout -- --nocapture`
- `cargo test --manifest-path candle-binding/Cargo.toml --no-default-features --lib test_bert_lora_ -- --nocapture`
- `cargo test --manifest-path candle-binding/Cargo.toml --no-default-features --lib model_architectures::traditional::bert_test -- --nocapture`
- `cargo check --manifest-path candle-binding/Cargo.toml --no-default-features`
- `git diff --check`

## Test Result

- Pooler regression test: passed, 1/1
- LoRA tests: passed, 8/8
- Traditional BERT tests: passed, 9/9
- `cargo check`: passed
- `git diff --check`: passed
- Real checkpoint tests were skipped because local model files are unavailable.
- `make test-binding-minimal` was attempted; `candle-binding` built successfully, but the gate stopped in unrelated `ml-binding` due to a Windows Rust compiler/linker ICE.
- `cargo fmt --check` could not run because `rustfmt` is unavailable in the local toolchain.

---
<details>
<summary>Semantic Router PR Checklist</summary>

- [x] PR title begins with exactly one bracketed category.
- [x] The title does not stack prefixes.
- [x] The PR links accepted issue #2249 with exactly one owner: `wg/router-models-inference-runtime`.
- [x] Commits in this PR are signed off with `git commit -s`.
- [x] The Purpose, Test Plan, and Test Result sections reflect the actual scope, commands, and blockers.

</details>

See [CONTRIBUTING.md](../CONTRIBUTING.md) for the full contributor workflow and commit guidance.

### CODE REVIEW

**Comment by Alanxtl** [comment]
the ci fail is not caused by the pr, maybe need to update the docker image or rerun it

**Comment by FAUST-BENCHOU** [comment]
> the ci fail is not caused by the pr, maybe need to update the docker image or rerun it

wait for https://github.com/vllm-project/semantic-router/issues/3312

**Review by Xunzhuo — APPROVED** [review]
Thanks for the focused fix, and welcome! The pooler shape correction and LoRA regressions cover the affected paths cleanly. LGTM.
